frontend/src/components/utils/Search.tsx

```diff
@@ -186,19 +186,9 @@ const Search: React.FC<Props> = React.memo(({ dashboard }) => {
 
       const cleanedSearchValue = value.replace(/\s/g, "");
 
-      let blockPromise: Promise<string | null>;
-      const maybeBlockHeight = cleanedSearchValue.replace(/[,]/g, "");
```

Comment:
No need for complicated logic on client - just check if it's a number server-side and return hash.